tests/Integration/Behaviour/Features/Context/Domain/Product/ProductImageFeatureContext.php

```diff
@@ -398,13 +404,77 @@ private function uploadImageByShopConstraint(string $imageReference, string $fil
     /**
      * @param string $productReference
      * @param TableNode $tableNode
-     * @param array $expectedShopIds
      *
      * @return void
      */
-    private function assertProductImagesForShops(string $productReference, TableNode $tableNode, array $expectedShopIds)
+    private function assertAllProductImages(string $productReference, TableNode $tableNode)
     {
         $images = $this->getProductImages($productReference);
+        $this->assertProductImages($tableNode, $images);
+    }
+
+    /**
+     * @param string $productReference
+     * @param TableNode $tableNode
+     * @param int $shopId
+     *
+     * @return void
+     */
+    private function assertProductImagesByShopId(string $productReference, TableNode $tableNode, int $shopId)
```

Comment:
minor thing, but missing everywhere
```suggestion
    private function assertProductImagesByShopId(string $productReference, TableNode $tableNode, int $shopId): void
```